File: app/src/main/java/cn/mycommons/androidmodular/app/ModuleLifeCycleManager.java
package cn.mycommons.androidmodular.app;

import android.app.Application;
import android.support.annotation.NonNull;

import java.util.ArrayList;
import java.util.List;

import cn.mycommons.modulebase.IModuleConfig;
import cn.mycommons.modulebase.IModuleLifeCycle;
import cn.mycommons.moduleorder.OrderModuleLifeCycle;
import cn.mycommons.moduleshopping.ShoppingModuleLifeCycle;
import cn.mycommons.moduleuser.UserModuleLifeCycle;

/**
 * ModuleLifeCycleManager <br/>
 * Created by xiaqiulei on 2017-05-14.
 */
class ModuleLifeCycleManager {

    @NonNull
    private ModuleConfig moduleConfig;
    @NonNull
    private final List<IModuleLifeCycle> moduleLifeCycleList;

    ModuleLifeCycleManager(@NonNull Application application) {
        moduleConfig = new ModuleConfig();
        moduleLifeCycleList = new ArrayList<>();
        moduleLifeCycleList.add(new UserModuleLifeCycle(application));
        moduleLifeCycleList.add(new OrderModuleLifeCycle(application));
        moduleLifeCycleList.add(new ShoppingModuleLifeCycle(application));
    }

    void onCreate() {
        for (IModuleLifeCycle lifeCycle : moduleLifeCycleList) {
            lifeCycle.onCreate(moduleConfig);
        }
    }

    void onTerminate() {
        for (IModuleLifeCycle lifeCycle : moduleLifeCycleList) {
            lifeCycle.onTerminate();
        }
    }

    @NonNull
    IModuleConfig getModuleConfig() {
        return moduleConfig;
    }
}

File: test-module/app/src/main/java/cn/mycommons/androidmodular/app/ModuleLifeCycleManager.kt
package cn.mycommons.androidmodular.app

import android.app.Application
import cn.mycommons.modulebase.IModuleConfig
import cn.mycommons.modulebase.IModuleLifeCycle
import cn.mycommons.moduleorder.OrderModuleLifeCycle
import cn.mycommons.moduleshopping.ShoppingModuleLifeCycle
import cn.mycommons.moduleuser.UserModuleLifeCycle
import java.util.*

/**
 * ModuleLifeCycleManager <br></br>
 * Created by xiaqiulei on 2017-05-14.
 */
internal class ModuleLifeCycleManager(application: Application) {

    private val moduleConfig: ModuleConfig = ModuleConfig()
    private val moduleLifeCycleList: MutableList<IModuleLifeCycle>

    init {
        moduleLifeCycleList = ArrayList()
        moduleLifeCycleList.add(UserModuleLifeCycle(application))
        moduleLifeCycleList.add(OrderModuleLifeCycle(application))
        moduleLifeCycleList.add(ShoppingModuleLifeCycle(application))
    }

    fun onCreate() {
        for (lifeCycle in moduleLifeCycleList) {
            lifeCycle.onCreate(moduleConfig)
        }
    }

    fun onTerminate() {
        for (lifeCycle in moduleLifeCycleList) {
            lifeCycle.onTerminate()
        }
    }

    fun getModuleConfig(): IModuleConfig {
        return moduleConfig
    }
}